### PR TITLE
Add coverage for repository functionality (BZ1383258)

### PR DESCRIPTION
**Pull request by oshtaier**
```
nosetests tests/foreman/cli/test_repository.py -m test_positive_create_with_mirror_on_sync
.
----------------------------------------------------------------------
Ran 1 test in 54.943s

OK
```

### CODE REVIEW

**Comment by abalakh** [comment]
@oshtaier since you've added new parameter, maybe it's worth implementing `update` test as well?

**Comment by oshtaier** [comment]
Updated
```
nosetests tests/foreman/cli/test_repository.py -m test_positive_update_mirror_on_sync
.
----------------------------------------------------------------------
Ran 1 test in 48.297s

OK
```

File: tests/foreman/cli/test_repository.py [hunk 1]
```diff
@@ -1077,6 +1098,26 @@ def test_positive_update_gpg_key(self):
...
+        """
+        new_repo = self._make_repository({
+            u'mirror-on-sync': 'no',
+        })
+        Repository.update({
+            u'id': new_repo['id'],
+            u'mirror-on-sync': 'yes',
+        })
+        result = Repository.info({'id': new_repo['id']})
+        self.assertEqual(result['mirror-on-sync'], 'yes')
```
Comment (abalakh) [inline tests/foreman/cli/test_repository.py]: nitpick: please use subtest in a similar to `create` test way and ensure mirror-on-sync can be switched back to 'no'.
We had issues with this parameter not being able to switch from one state to another correctly; it's not the case here as you aren't really verifying how it's working, but imo it's not completely useless and worth covering.

File: tests/foreman/cli/test_repository.py [hunk 2]
```diff
@@ -1077,6 +1098,26 @@ def test_positive_update_gpg_key(self):
...
+        """Update the mirror on sync rule for repository
+
+        :id: 9bab2537-3223-40d7-bc4c-a51b09d2e812
+
+        :expectedresults: Repository is updated
+
+        :CaseImportance: Critical
+        """
+        new_repo = self._make_repository({
+            u'mirror-on-sync': 'no',
```
Comment (abalakh) [inline tests/foreman/cli/test_repository.py]: nitpick:
```python
        new_repo = self._make_repository({'mirror-on-sync': 'no'})
```

**Comment by oshtaier** [comment]
Addressed all comments till that moment

**Pull request merged**